detector/cve/cve20258088/cve20258088.go

```diff
@@ -0,0 +1,331 @@
+	err = peFile.Parse()
+	if err != nil {
+		log.Debugf("cve20258088: extractPEVersion: failed to parse PE file: %s, error: %v", exePath, err)
+		// Fallback to filename heuristics
+		return extractVersionFromFilename(exePath)
+	}
+
+	// Extract version resources
+	versionInfo, err := peFile.ParseVersionResources()
+	if err != nil {
+		log.Debugf("cve20258088: extractPEVersion: failed to parse version resources: %s, error: %v", exePath, err)
+		// Fallback to filename heuristics
+		return extractVersionFromFilename(exePath)
+	}
+
+	// Extract version and product name from version resources
+	if productVersion, ok := versionInfo["ProductVersion"]; ok && productVersion != "" {
+		version = productVersion
+	} else if fileVersion, ok := versionInfo["FileVersion"]; ok && fileVersion != "" {
+		version = fileVersion
+	}
+
+	if productName, ok := versionInfo["ProductName"]; ok && productName != "" {
+		prodName = productName
+	} else if internalName, ok := versionInfo["InternalName"]; ok && internalName != "" {
+		prodName = internalName
+	} else if originalFilename, ok := versionInfo["OriginalFilename"]; ok && originalFilename != "" {
+		prodName = originalFilename
+	}
+
+	// If we couldn't extract from PE resources, fallback to filename heuristics
+	if version == "" || prodName == "" {
+		fallbackVersion, fallbackProdName := extractVersionFromFilename(exePath)
+		if version == "" {
+			version = fallbackVersion
+		}
+		if prodName == "" {
+			prodName = fallbackProdName
+		}
+	}
+	return version, prodName
+}
+
+// extractVersionFromFilename tries to extract version info and product name from filename (fallback method)
+func extractVersionFromFilename(exePath string) (version, prodName string) {
+	base := filepath.Base(exePath)
+	lower := strings.ToLower(base)
+
+	switch {
+	case strings.Contains(lower, "winrar"):
+		prodName = "WinRAR"
+	case strings.Contains(lower, "unrar"):
+		prodName = "UnRAR"
+	case strings.Contains(lower, "rar"):
+		prodName = "RAR"
+	}
+
+	archRegex := regexp.MustCompile(`(?i)(x64|x86|arm64|arm32|portable)`)
+	clean := archRegex.ReplaceAllString(lower, "")
+
+	clean = strings.TrimSuffix(clean, ".exe")
+	clean = strings.TrimSuffix(clean, ".dll")
+
+	// Find version number (supports 610, 6.10, 6_10, etc.)
+	verRegex := regexp.MustCompile(`(\d+[\._-]?\d*)`)
+	verMatch := verRegex.FindStringSubmatch(clean)
+	if verMatch != nil {
+		verStr := verMatch[1]
+		// Normalize: 610 -> 6.10, 623 -> 6.23, etc.
+		if len(verStr) == 3 && !strings.ContainsAny(verStr, "._-") {
+			version = fmt.Sprintf("%s.%s", verStr[:1], verStr[1:])
+		} else if len(verStr) == 4 && !strings.ContainsAny(verStr, "._-") {
+			version = fmt.Sprintf("%s.%s", verStr[:2], verStr[2:])
+		} else {
+			version = strings.ReplaceAll(verStr, "_", ".")
+			version = strings.ReplaceAll(version, "-", ".")
+		}
+	}
+	return version, prodName
+}
+
+// isAffectedVersion returns true if version < 7.13
+func isAffectedVersion(ver string) bool {
+	parts := strings.Split(ver, ".")
```

Comment:
We have a built-in version comparison library in SCALIBR: https://github.com/google/osv-scalibr/blob/396e8cfb62cd9c0a439b46a12406d1677292e0eb/semantic/parse.go#L43

You can do something like semantic.Parse(ver).CompareStr("7.13")